docs/features/populate.rst

```diff
@@ -0,0 +1,296 @@
+
+      vars:
+        admin_username: admin
+        admin_password: changeme
+        org_name_list:
+          - company7
+          - company8
+        prefix: aaaa
+        suffix: bbbb
+        my_name: me
+
+actions
++++++++
+
+The actions is the most important section of the YAML, it is a list of actions
+being each action a dictionary containing special keys depending on the action type.
+
+The action type is defined in ``action`` key and available actions are:
+
+Actions are executed in the defined order and order is very important because
+each action can ``register`` its result to the internal registry to be referenced
+later in any other action.
+
+
+**CRUD ACTIONS**
+
+Crud actions takes a ``model`` argument, any from ``nailgun.entities`` is valid.
+
+
+- create (the default)
+
+  Creates a new entity if not exists, else gets existing.
+- update
+
+  Updates entity with provided ``data`` by ``id`` or unique search
+- delete
+
+  deleted entity with ``id`` or unique search
+
+**SPECIAL ACTIONS**
+
+- echo
+
+  Logs and print output to the console
+- register
+
+  Register a variable in the internal registry
+- unregister
+
+  removes a variable from register
+- assertion
+
+  perform assertion operations, if any fails returns exit code 1
+
+Dynamic Data
+++++++++++++
+
+There are some ways to fetch dynamic data in action definitions, it depends
+on the action type.
+
+For any key you can use ``Jinja`` to provide a dynamic value as in::
+
+  value: "{{ get_something }}"
+  value: "{{ fauxfactory.gen_string('alpha')}}"
+  value: user_{{ item }}
+
+For some actions you can provide a ``data`` key, that data is used to create
+new entities and also to perform searches or build the action function.
+
+Every ``data`` key accepts 4 special reference directives in its sub-keys.
+
+- from_registry
+
+  Gets anything from registry::
+
+    data:
+      organization:
+        from_registry: default_org
+      name:
+        from_registry: my_name
+
+- from_object
+
+  Gets any Python object available in the environment::
+
+    data:
+      url:
+        from_object:
+          name: robottelo.constants.FAKE_0_YUM_REPO
+
+- from_search
+
+  Perform a search and return its result::
+
+    data:
+      organization:
+        from_search:
+          model: Organization
+          data:
+            name: Default Organization
+
+- from_read
+
+  Perform a read operation, which is useful when we have unique data or id::
+
+    data:
+      organization:
+        from_read:
+          model: Organization
+          data:
+            id: 1
+
+
+The internal registry
++++++++++++++++++++++
+
+Every action which returns a result can write its result to the registry, so
+it is available to be accessed by other actions.
+
+Provide a ``register`` unique name in ``action`` definition.
+
+The actions that support ``register`` are:
+
+- create
+- update
+- register
+- assertion
+
+All dynamic directives ``from_*`` supports the use of ``register``
```

Comment:
This is not clear to me what it means. It means that I can have something like:

```
from_registry: some_registry_var
    register: my_reg
```

and that will copy what is in some_registry_var to my_reg?

It would be great to expand a little bit about this or add an example.